File: sentimental-cash/cash.py
from cs50 import get_float


while True:
    change = get_float("Change: ")  # ask user for change in dollars
    if change > 0:  # change has to be more than 0
        break


def calculate_quarters(change):
    quarters = 0
    while (change >= 25):
        quarters += 1
        change = change - 25
    return quarters


def calculate_dimes(change):
    dimes = 0
    while (change >= 10):
        dimes += 1
        change = change - 10
    return dimes


def calculate_nickels(change):
    nickels = 0
    while (change >= 5):
        nickels += 1
        change = change - 5
    return nickels


def calculate_pennies(change):
    pennies = 0
    while (change >= 1):
        pennies += 1
        change = change - 1
    return pennies


change = change * 100

quarters = calculate_quarters(change)  # how many quarters
change = change - (quarters * 25)  # subtract value of quarters

dimes = calculate_dimes(change)  # how many dimes
change = change - (dimes * 10)  # subtract value of dimes

nickels = calculate_nickels(change)  # how many nickels
change = change - (nickels * 5)  # subtract value of nickels

pennies = calculate_pennies(change)  # how many dimes
change = change - (pennies * 1)  # subtract value of dimes

coins = quarters + dimes + nickels + pennies
print(coins)

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int change);
int calculate_dimes(int change);
int calculate_nickels(int change);
int calculate_pennies(int change);

int main(void)
{
    // Prompt for change owed in cents
    int change;
    do
    {
        change = get_int("Amount of change owed, in cents: ");
    }
    while (change < 0);

    // How many quarters
    int quarters = calculate_quarters(change);
    // Subtract the value of the quarters
    change = change - (quarters * 25);

    // How many dimes
    int dimes = calculate_dimes(change);
    // Subtract the value of the dimes
    change = change - (dimes * 10);

    // How many nickels
    int nickels = calculate_nickels(change);
    // Subtract the value of the nickels
    change = change - (nickels * 5);

    // How many pennies
    int pennies = calculate_pennies(change);
    // Subtract the value of the pennies
    change = change - (pennies * 1);

    // Add up coins used
    int coins = quarters + dimes + nickels + pennies;

    // Print sum
    printf("%i\n", coins);
}

int calculate_quarters(int change)
{
    // Divide by 25c
    int quarters = 0;
    while (change >= 25)
    {
        quarters++;
        change = change - 25;
    }
    return quarters;
}

int calculate_dimes(int change)
{
    // Divide remainder by 10c
    int dimes = 0;
    while (change >= 10)
    {
        dimes++;
        change = change - 10;
    }
    return dimes;
}

int calculate_nickels(int change)
{
    // Divide remainder by 5c
    int nickels = 0;
    while (change >= 5)
    {
        nickels++;
        change = change - 5;
    }
    return nickels;
}

int calculate_pennies(int change)
{
    // Divide remainder by 1c
    int pennies = 0;
    while (change >= 1)
    {
        pennies++;
        change = change - 1;
    }
    return pennies;
}
